src/transactions/transferFeedUtils.ts

```diff
@@ -400,3 +426,56 @@ export function isTransferTransaction(
 ): tx is TransferItemFragment {
   return (tx as TransferItemFragment).address !== undefined
 }
+
+// Note: This hook is tested from src/transactions/feed/TransferFeedItem.test.ts
+function useFiatConnectTransferDisplayInfo({ amount, transactionHash }: TokenTransfer) {
+  const { t } = useTranslation()
+  const tokenInfo = useTokenInfo(amount.tokenAddress)
+  const fcTransferDetails = useSelector(getCachedFiatConnectTransferSelector(transactionHash))
+  const cachedFiatAccountUses = useSelector(cachedFiatAccountUsesSelector)
+  const account = useMemo(
+    () =>
+      fcTransferDetails?.quote.transferType === TransferType.TransferOut
+        ? cachedFiatAccountUses.find(
+            ({ fiatAccountId }) => fiatAccountId === fcTransferDetails.fiatAccountId
+          )
+        : undefined,
+    [cachedFiatAccountUses, fcTransferDetails]
+  )
+  if (!account || !fcTransferDetails) {
+    if (fcTransferDetails?.quote.transferType === TransferType.TransferIn) {
+      Logger.debug(TAG, 'useFiatConnectTransferDisplayInfo only supports transfers out (withdraws)')
+    }
+    return
+  }
+
+  const fiatAmount = Number(fcTransferDetails.quote.fiatAmount)
+  const cryptoAmount = Number(amount.value)
+  const exchangeRate = String(fiatAmount / cryptoAmount)
+  const localAmount: LocalAmount = {
+    //Negative sign because currently only withdraws are supported
+    value: -new BigNumber(fiatAmount),
+    currencyCode: convertToLocalCurrency(fcTransferDetails.quote.fiatType),
+    exchangeRate,
+  }
```

Comment:
```suggestion
  const fiatAmount = new BigNumber(fcTransferDetails.quote.fiatAmount)
  const localAmount: LocalAmount = {
    //Negative sign because currently only withdraws are supported
    value: fiatAmount.multipliedBy(-1),
    currencyCode: convertToLocalCurrency(fcTransferDetails.quote.fiatType),
    exchangeRate: fiatAmount.div(amount.value).toFixed(2),
  }
```
my previous comment was less about why BigNumber is used for "value", and more about why it *isn't* used for the arithmetic used in calculating exchangeRate. We should all be scared of [floating point errors in javascript](https://ellenaua.medium.com/floating-point-errors-in-javascript-node-js-21aadd897bf8)

Reply:
I see, yeah that is a safer way to calculate the exchangeRate